examples_tests/testbed.py

```diff
@@ -0,0 +1,112 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015, 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import platform
+import subprocess
+import time
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_snappy_image(output_directory):
+    logger.info('Creating a snappy image to run the tests.')
+
+    image_path = os.path.join(output_directory, 'snappy.img')
+    subprocess.check_call(
+        ['sudo', 'ubuntu-device-flash', '--verbose',
+         'core', '15.04', '--channel', 'stable',
+         '--output', image_path, '--developer-mode'])
```

Comment:
check_call will raise an exception if the return code is different than 0. That will stop the test and show the traceback, which I think it's enough.